andela/codepirates-ah-backend
# #167313397 Email verification

#### What does this PR do?

- Verifies if the email used by the user when signing up or admin creating other admins exists.

#### Description of Task to be completed?

- Created the email template that has to be sent to the user registering to Authors' heaven.

- Created a url (route) that verifies the user.

- Created a middleware that prevent a user to access other routes when he/she didn't confirm the email.

#### How should this be manually tested?

1. Clone the repo

2. Run <npm install>

3. Create .env file, and  include the credentials found in .env.sample file

4. Run <npm run migration>

5. Use postman to test sign up 

6. Go the email used when signing up and verify the received email.

#### Any background context you want to provide?

- none

#### What are the relevant pivotal tracker stories?

 [167313397](https://www.pivotaltracker.com/story/show/167313397)

#### Screenshots (if appropriate)

<img width="1432" alt="Screen Shot 2019-08-08 at 12 10 46" src="https://user-images.githubusercontent.com/45101716/62695061-dd17ae00-b9d5-11e9-91e8-dda7fa9b1392.png">

<img width="1430" alt="Screen Shot 2019-08-08 at 12 11 39" src="https://user-images.githubusercontent.com/45101716/62695070-e1dc6200-b9d5-11e9-9494-5320e6b76e7f.png">


<img width="131" alt="Screen Shot 2019-08-08 at 12 10 33" src="https://user-images.githubusercontent.com/45101716/62695108-f28cd800-b9d5-11e9-89e4-7d2d8c0eb118.png">


<img width="400" alt="Screen Shot 2019-08-08 at 12 12 05" src="https://user-images.githubusercontent.com/45101716/62695123-f91b4f80-b9d5-11e9-81e2-e1769f7c64e7.png">

#### Questions:

@Mnickii @Anguandia @minega25 @nshutijonathan @nkalyesubula @salviosage
